### DISCUSSION TITLE
How to group test steps to mark transactions in a artillery + playwright ui workflow scenario

### DISCUSSION THREAD

**Discussion by LahiruMadhawa2020** [artilleryio/artillery#1768] [Artillery]
Hi Team,

In artillery + playwright ui workflow scenario, we are trying to test an end to end performance test journey. To get the accurate response times of the corresponding transactions (user actions), we need to group the steps in to transactions. Really appreciate if you can let me know how to mark the transactions/grouping in artillery tool.

Thank you in advance.

**Comment by hassy** [comment]
hi @LahiruMadhawa2020 👋 We have an example of grouping several Playwright actions into a transaction and reporting combined response times here:

https://github.com/artilleryio/artillery/tree/main/examples/browser-load-testing-playwright#example-3-tracking-custom-metrics-for-part-of-the-flow

Does that work for your use-case?

**Reply by LahiruMadhawa2020** [reply]
Hi @hassy, 
Thank you very much for your response and sorry for noticing the response late. 
I looked into the answer and yes, the use case is covering several transactions. However, what I really want is to separate the results by grouping into specific transactions.
From the given example what I want is to separate the results related to different metrics belong to specific transactions (ie: 1. Go to the homepage, 2. Search for a product, ...etc.).
Thank you

**Comment by SheshashayanPapanna** [comment]
@LahiruMadhawa2020, could you please let me know if you succeeded finding a solution or work around to group user actions into separate transactions and obtain the corresponding response times ?
Thanks in advance!

**Reply by bernardobridge** [reply]
Hi @SheshashayanPapanna and @LahiruMadhawa2020  👋 

We released this functionality https://www.artillery.io/docs/reference/engines/playwright#teststep-argument recently, which I believe is what you're looking for? It would allow you to wrap parts of your Playwright code (user actions) into steps, and then emits timers for those.

It won't be the response times of any API calls the browser is making, but rather the time taken in the UI to do each step (which can generate multiple network calls).

Let us know if that doesn't work!
